engine/gameobject/src/gameobject/gameobject_script.cpp

```diff
@@ -673,99 +750,108 @@ namespace dmGameObject
         }
         dmGameObject::PropertyDesc property_desc;
         dmGameObject::PropertyResult result = dmGameObject::GetProperty(target_instance, target.m_Fragment, property_id, property_options, property_desc);
+
+        if (result == dmGameObject::PROPERTY_RESULT_OK && !index_requested && property_desc.m_ValueType == dmGameObject::PROP_VALUE_ARRAY && property_desc.m_ArrayLength > 0)
+        {
+            lua_newtable(L);
+
+            // We already have the first value, so no need to get it again
+            HandleGoGetResult(L, result, property_desc, property_id, target_instance, target, property_options, index_requested);
```

Comment:
A little mismatch with the api here, since you're not receiving the "lua_error" result.
I know, we don't want to really catch it, but I think it speaks to readability.
I think the biggest loss is that it's not apparent that the code might not pass beyond this line of code if an error occurs.